typescript/listeners/format.py

```diff
@@ -11,5 +13,35 @@ def on_post_text_command_with_info(self, view, command_name, args, info):
              "typescript_paste_and_format"]:
             print("handled changes for " + command_name)
 
+    def on_modified_with_info(self, view, info):
+        log.debug("Format on key")
+
+        if (
+            is_typescript(view) and
+            cli.ts_auto_format_enabled and
+            info.prev_sel and
+            len(info.prev_sel) == 1 and
+            info.prev_sel[0].empty()
+        ):
+            last_command, args, repeat_times = view.command_history(0)
+            log.debug("last_command:{0}, args:{1}".format(last_command, args))
+            if last_command == "insert":
+                pos = info.prev_sel[0].begin()
+                if ";" in args["characters"]:
+                    view.run_command("typescript_format_on_key", {"key": ";"})
+                if "}" in args["characters"]:
+                    if cli.auto_match_enabled:
+                        prev_char = view.substr(pos - 1)
+                        post_char = view.substr(pos + 1)
```

Comment:
if you are at point before the beginning of the buffer or after the ending of the buffer, ``view.substr`` won't throw exception, it will simply return `\x00`. Which is fine because they wouldn't be equal to `{` or `}` to interfere the conditions.